packages/alfa-rules/src/common/dom/get-colors.ts

```diff
@@ -276,6 +276,17 @@ export function getForeground(
       const style = Style.from(element, device, context);
       const foregroundColor = style.computed("color").value;
 
+      if (
+        foregroundColor.type === "keyword" &&
+        foregroundColor.value === "currentcolor" &&
+        element.parent().isSome()
+      ) {
+        const parent = element.parent().get();
+        if (isElement(parent)) {
+          return getForeground(parent, device, context);
+        }
+      }
+
```

Comment:
Thinking about it, I'm getting scared this may mess up opacity resolution (since the recursive call to `getForeground` will resolve the `opacity` of the parent but likely shouldn't, i.e.
```html
<div style="color: blue; opacity: 0.5">
  <span style="color: currentcolor">Hello</span>
</div>
```
The `<span>` should have a computed `color` of `blue` (which is them resolved with opacity), but `getForeground` on the parent returns 50% blue only.

---
So this should be more something like
```typescript
while(element.isSome() && !isCurrentColor(color)) {
  color = Style.from(element.get()).computed("color")
  element = element.parent().filter(isElement)
}
```